### DISCUSSION TITLE
How can I ping URL with JavaScript

### DISCUSSION THREAD

**Discussion by golden-hero** [community/community#45720] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

I need to check the URL before loading the website like `ping`.
Is there any good way to ping with JavaScript?

**Comment by eFastRun** [comment]
You can reference `ping-url` and `ping.js` npm module.

**Comment by eFutureDev** [comment]
I tried with `is-reachable` module and this was good for me.

**Comment by Krieit** [comment, marked as answer]
Javscript itself doesn't have a native way of sending a "ping", so results may not be completely accurate. Since ajax requests are affected by cross-domain issues (CORS), they are blocked by default. ping.js is using a method of loading a favicon.ico image from any host and timing the response time.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @golden-hero ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
